### PR TITLE
Add py1e warning when Shuffle block size is smaller than shard size

### PR DESCRIPTION
**Pull request by snarayan21**
## Description of changes:
Previously, when shuffle block size was less than the span size aka shard part size, the run would crash at [this line](https://github.com/mosaicml/streaming/blob/6b638518294f7338f7bb54e9bb529bfad8e5fa50/streaming/base/shuffle/py1e.py#L102) with `ValueError: high - low < 0` because the new shard range would not be defined. Instead, when this is the case, we now set the shard range extension to 0, meaning the shard range stays the same, so the shard is not shuffled with other neighboring shards at all. Because this is likely not the intended behavior, we warn the user when this happens as well. 

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [ ] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [ ] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [ ] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [ ] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [ ] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#submitting-a-contribution))
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: streaming/base/shuffle/py1e.py [hunk 1]
```diff
@@ -107,6 +116,12 @@ def get_shuffle_py1e(shard_sizes: NDArray[np.int64],
             # Update sample offset for the next shard.
             cn_sample_offset += span_size
 
+        # If warn_user is true, this means the block size for shifts was smaller than a span size.
+        # This will result in no shuffling being done on that span aka shard part, so warn user.
+        if warn_user:
+            warnings.warn('Shuffle block size was smaller than shard size for some shards. This \
```
Comment (karan6181) [inline streaming/base/shuffle/py1e.py:122]: Is there a way to validate this behavior using a unit test?

Comment (snarayan21) [reply streaming/base/shuffle/py1e.py:122]: Oh yeah let me add that!

Comment (snarayan21) [reply streaming/base/shuffle/py1e.py]: added unit test! I didn't even know pytest could detect warnings...

File: tests/test_streaming.py [hunk 2]
```diff
@@ -510,6 +510,37 @@ def run_one_iter(local: str, remote: str, seed: int) -> None:
...
+                               batch_size=batch_size,
+                               num_canonical_nodes=1,
+                               shuffle_seed=seed,
+                               shuffle_algo='py1e',
+                               shuffle_block_size=shuffle_block_size)
+    dataloader = DataLoader(dataset=dataset, batch_size=batch_size)
+
+    with pytest.warns(UserWarning, match=f'Shuffle block size was smaller than shard size*'):
+        for _ in dataloader:
+            print('yo')
```
Comment (karan6181) [inline tests/test_streaming.py:541]:
```suggestion
            pass
```

**Pull request merged**